### PR TITLE
add gatekeeper v3.11

### PR DESCRIPTION
**Pull request by thomasmckay**
Slight modifications to values.yaml to allow install on OCP-4.12. 

Required SCC prior to helm install:
```
oc new-project gatekeeper-system
oc create clusterrole gatekeeper-allow-anyuid-scc --verb=use --resource=securitycontextconstraints.security.openshift.io --resource-name=anyuid
oc create rolebinding gatekeeper-anyuid-scc --serviceaccount=gatekeeper-system:gatekeeper-admin --clusterrole=gatekeeper-allow-anyuid-scc -n gatekeeper-system
oc create rolebinding gatekeeper-upgrade-crds-anyuid-scc --serviceaccount=gatekeeper-system:gatekeeper-admin-upgrade-crds --clusterrole=gatekeeper-allow-anyuid-scc -n gatekeeper-system
oc create rolebinding gatekeeper-update-namespace-label-anyuid-scc --serviceaccount=gatekeeper-system:gatekeeper-update-namespace-label --clusterrole=gatekeeper-allow-anyuid-scc -n gatekeeper-system
```

### CODE REVIEW

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        annotations: {}
+        kubeversion: ""
+        dependencies: []
+        type: ""
+    chart-overrides: ""
+results:
+    - check: v1.0/not-contains-crds
+      type: Mandatory
+      outcome: FAIL
+      reason: Chart contains CRDs
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:42]: upstream contains CRDs

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - check: v1.0/chart-testing
+      type: Mandatory
+      outcome: FAIL
+      reason: 'Chart Install failure: Kubernetes cluster unreachable: Get "https://a8b551ebd76cb486aaf2bf3f505f4ecd-3f49ba7ef96bdbc9.elb.us-east-1.amazonaws.com:6443/version": dial tcp: lookup a8b551ebd76cb486aaf2bf3f505f4ecd-3f49ba7ef96bdbc9.elb.us-east-1.amazonaws.com: no such host'
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:46]: verified manually

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    - check: v1.0/contains-values-schema
+      type: Mandatory
+      outcome: FAIL
+      reason: Values schema file does not exist
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:50]: doesn't exist upstream

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    - check: v1.1/has-kubeversion
+      type: Mandatory
+      outcome: FAIL
+      reason: Kubernetes version is not specified
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:54]: why is this required?

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      outcome: PASS
+      reason: API version is V2, used in Helm 3
+    - check: v1.0/not-contain-csi-objects
+      type: Mandatory
+      outcome: PASS
+      reason: CSI objects do not exist
+    - check: v1.0/required-annotations-present
+      type: Mandatory
+      outcome: FAIL
+      reason: 'Missing required annotations: [charts.openshift.io/name]'
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:74]: why is this required?

File: charts/redhat/redhat/gatekeeper/3.11/report.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      outcome: SKIPPED
+      reason: 'Chart is not signed : Signature verification not required'
+    - check: v1.1/images-are-certified
+      type: Mandatory
+      outcome: SKIPPED
+      reason: 'Image certification skipped : registry.redhat.io/gatekeeper/gatekeeper-rhel8:v3.11'
+    - check: v1.0/contains-test
+      type: Mandatory
+      outcome: FAIL
+      reason: Chart test files do not exist
```
Comment (thomasmckay) [inline charts/redhat/redhat/gatekeeper/3.11/report.yaml:86]: no tests upstream

**Comment by komish** [comment]
o/ We discussed that you'll look to submit this as a community chart for the moment. Just wanted to note that the automation is currently complaining because there is not an existing directory structure for your project.

Please see the preparation steps described here for how to get started https://github.com/openshift-helm-charts/charts/tree/main/docs#preparation

**Comment by thomasmckay** [comment]
@komish Do you create the OWNERS file or do I? The docs are confusing; I'm not a partner but through Red Hat.

**Comment by thomasmckay** [comment]
@komish moved to community and re-ran chart-verifier

**Comment by komish** [comment]
@thomasmckay sorry for the delay. I've merged your owner's file. Rebase this PR and run chart verifier to start the certification workflow.

**Comment by thomasmckay** [comment]
> @thomasmckay I'm rerunning the failed CI. Right now the current complaint is that you've provided both the tarball containing charts and the chart directory - we only need one or the other.

I'd prefer to have both. The `tgz` is opaque and having the `src` dir lets consumers see the details. A better experience in this user's opinion. If I remove the `tgz` will it still be installable? I don't know the implications of one or the other.

**Comment by thomasmckay** [comment]
Related to above, can I include a Makefile in here as well? This repo is just a sync with another source and some automation to copy, run chart-verify, etc. would be useful to maintainers.

**Comment by komish** [comment]
@thomasmckay just FYI, Our bot is telling you that your resource is certified, but it's due to a bug we'll investigate. 

The issue you currently have is that both the tarball and src coexist. Pick whichever you want to leave, run chart-verifier against that, submit the report, and we'll have CI run again once you submit. 

My apologies for the bot's bugs.

**Comment by thomasmckay** [comment]
@komish removed the tgz in favor of just src

**Comment by thomasmckay** [comment]
which dir can i add files to, such as README.md, Makefile, etc.?

**Comment by komish** [comment]
@thomasmckay it would likely have to be your chart directory itself.

**Comment by komish** [comment]
You're certifying as a community chart, so you need to use chart-verifier with the `--set profile.vendorType=community` option.

**Comment by komish** [comment]
@thomasmckay I've got just one more thing I need to check with some folks before I can merge this. Nothing for you to do at the moment. Thanks for your effort here. I'll get back to you as soon as I can.
